### ISSUE TITLE
storage space issues

### ISSUE THREAD

**Issue by SuiJiGuoChengSuiJiGuo** [rraileanu/policy-dynamics-value-functions#2]
Hello, I met some problems when I run your code, and I would like to ask you:
1. I encountered the problem of memory explosion when running the train_policy_embedding.py file (according to the parameters in README). We have 256G of memory (about 130G available) and 20G of video memory. So whether the memory required in this part is too large.
2. The embedding training code needs to re-collect data every time, will it affect the final performance if the data is collected and stored in advance and reused?
3. What is the horizon of each episode of the experiment in Figure 4 in the paper? Take ant-wind environment as an example, the steps of episodes in env_utils.py  is 256 steps, but the returns of ppo running 256 steps seems to reach only 200-300.

**Comment by rraileanu** [comment]
Hi! Thank you for the questions. 

1. Do you get a memory error when loading the pre-trained policies or when you run the policy embedding model? You could try using fewer seeds (i.e. only 2 or 3) to train the policy embedding model. I don't think it should affect performance too much. You could also try to collect the data requires to train the policy embedding beforehand and use that dataset to train the embeddings, without loading the policy models.

2. No, I don't think this should change the results very much.

3. For Swimmer the maximum number of steps is 50, while for the MuJoCo environments, it is the default of the original environments i.e. 256 for the Ant environments. Note that the environments used for evaluation in Figure 4 are modified versions of the original MuJoCo environments which means that they are more challenging than the typical Ant / Swimmer tasks, so a PPO agent gets less reward.

Let me know if you have any other questions.

**Comment by SuiJiGuoChengSuiJiGuo** [comment]
Hello, thank you very much for your suggestion.
We tested based on the parameters you gave in the readme.md, but the results we got were about 20% worse than the results in the paper, so I would like to ask if there are some other parameters we did not tune properly?
Also, we have a question about the "train_policy_embedding.py". The code loaded all policies in line 48, but when interacting with the environment (line 108), it seems that only the first 15 of these policies are used? Does this have some effect on the final result?

**Comment by rraileanu** [comment]
Hi! That sounds like a bug. It should be training on all loaded policies. Perhaps that can explain the difference in the results. 

Are you running multiple seeds and looking at the average? There can be quite a bit of variance in the final reward.
